metrics-exporter/src/bin/io_engine/cache/pool_stat.rs

```diff
@@ -0,0 +1,45 @@
+use super::{Cache, ResourceOps};
+use crate::client::{
+    grpc_client::GrpcClient,
+    pool_stat::{PoolIoStat, PoolIoStats, PoolStatOperations},
+};
+use std::ops::DerefMut;
+use tracing::error;
+
+impl ResourceOps for PoolIoStats {
+    type ResourceVec = Vec<PoolIoStat>;
+
+    fn set(&mut self, val: Self::ResourceVec) {
+        self.pool_stats = val
+    }
+
+    fn invalidate(&mut self) {
+        self.pool_stats = vec![]
+    }
+}
+
+/// To store pool iostat data in cache.
+pub(crate) async fn store_pool_stats_data(client: &GrpcClient) -> Result<(), ()> {
+    let pool_stats = client.get_pool_iostat().await;
+    let mut cache = match Cache::get_cache().lock() {
+        Ok(cache) => cache,
+        Err(error) => {
+            error!(%error, "Error while getting cache resource");
+            return Err(());
+        }
+    };
+    let pools_cache = cache.deref_mut();
```

Comment:
@tiagolobocastro , We call deref_mut() here. then `pool_cache` is used to `set` or `invalidate` mutably.

Reply:
This is the bit I don't get, I don't see the DerefMut trait being implemented, what is this deref_mut returning?

Reply:
Cache is OnceCell which has Mutex<Cache>. When we acquire lock on Mutex. We get `MutexGuard`. This implements DerefMut/ Deref.

Reply:
Nevermind, I misread, we're matching n the pool_stats and not the cache :)